File: backend/Models/Submission.cs
namespace Learnova.Models;

public enum SubmissionStatus
{
    Submitted,
    Graded
}

public class Submission
{
    public Guid Id { get; set; }
    public Guid AssignmentId { get; set; }
    public Assignment Assignment { get; set; } = null!;
    public Guid StudentId { get; set; }
    public User Student { get; set; } = null!;
    public required string Answer { get; set; }
    public required DateTime SubmittedAt { get; set; }
    public int? Marks { get; set; }
    public string? Feedback { get; set; }
    public SubmissionStatus SubmissionStatus { get; set; } = SubmissionStatus.Submitted;

}

File: frontend/src/types/submission.ts
export type SubmissionStatus = 'Submitted' | 'Graded';

export interface Submission {
  id: string;
  assignmentId: string;
  assignmentTitle: string;
  studentId: string;
  studentName: string;
  answer: string;
  submittedAt: string;
  marks: number | null;
  feedback: string | null;
  submissionStatus: SubmissionStatus;
}

export interface SubmissionGradeFormData {
  marks: number;
  feedback: string;
}

export interface SubmissionFormData {
  answer: string;
}
